channel.go

```diff
@@ -50,27 +54,39 @@ type Channel struct {
 	client *Client
 }
 
-type queryResponse struct {
+type channelQueryResponse struct {
 	Channel  *Channel         `json:"channel,omitempty"`
 	Messages []*Message       `json:"messages,omitempty"`
 	Members  []*ChannelMember `json:"members,omitempty"`
 	Read     []*ChannelRead   `json:"read,omitempty"`
 }
 
-func (q queryResponse) updateChannel(ch *Channel) {
+func (ch *Channel) update(q channelQueryResponse) {
+	// atomically add or create new lock in channel lock store
+	lock, _ := channelsLock.LoadOrStore(ch.CID, &sync.Mutex{})
```

Comment:
Yikes! 😅
```go
ch.mutex.Lock()
refer ch.mutex.Unlock()
```

You could dedupe Channel instantiation (with a sync.Map) if really needed.

Reply:
if lock will be stored in Channel struct, updating channel will be more complex

Reply:
I don't understand how placing the muted in the respective struct causes updates to be more complex? Just remove the map and you're good to go.